src/main/java/plugily/projects/thebridge/events/Events.java

```diff
@@ -239,21 +243,31 @@ public void onInteractWithArmorStand(PlayerArmorStandManipulateEvent event) {
 
   @EventHandler
   public void onCraft(PlayerInteractEvent event) {
-    if(!ArenaRegistry.isInArena(event.getPlayer())) {
-      return;
-    }
+    if(!ArenaRegistry.isInArena(event.getPlayer())) return;
+
     if(event.getPlayer().getTargetBlock(null, 7).getType() == XMaterial.CRAFTING_TABLE.parseMaterial()) {
       event.setCancelled(true);
     }
   }
 
   @EventHandler
   public void onPlayerTeleport(PlayerTeleportEvent event) {
-    if(!ArenaRegistry.isInArena(event.getPlayer())) {
-      return;
-    }
+    if(!ArenaRegistry.isInArena(event.getPlayer())) return;
+
     if(event.getCause().equals(PlayerTeleportEvent.TeleportCause.END_PORTAL)) {
       event.setCancelled(true);
     }
   }
+
+  @EventHandler
+  public void WeatherChangeEvent(WeatherChangeEvent event) {
+    if (!event.toWeatherState()) return;
+    if (!(Objects.equals(ConfigUtils.getConfig(plugin, "config").getString("NoWeather"), "true"))) return;
+
+    if (ConfigUtils.getConfig(plugin, "config").getStringList("NoWeather-Worlds").contains(event.getWorld().getName())) {
+      event.setCancelled(true);
+      event.getWorld().setWeatherDuration(0);
+      event.getWorld().setThundering(false);
```

Comment:
It would make more sense to get the world of the arena and set the weather to clear there